test/abstractServer.cpp

```diff
@@ -79,6 +82,18 @@ void AbstractServer::Run() {
     server_stop_cv.notify_all();
 }
 
+void AbstractServer::AddConnection(int remote_port) {
+    unique_connections.insert(remote_port);
```

Comment:
Should we take care of cleaning them up once the connection is shut down?

Reply:
The test relies on checking the uniq connection after connections might be closed, instead it does manual cleanup of the state after confirming the count.